libsolidity/interface/StandardCompiler.cpp

```diff
@@ -623,8 +623,135 @@ std::variant<OptimiserSettings, Json::Value> parseOptimizerSettings(Json::Value
 	return { std::move(settings) };
 }
 
+/// Only parses the input settings and fills the InputsAndSettings with it.
+/// If this fails, it does return a JSON error response, otherwise nullopt on success.
+optional<Json::Value> parseInputSettings(Json::Value _settings, StandardCompiler::InputsAndSettings& _output)
```

Comment:
@chriseth's comments from the old PR that are still relevant (https://github.com/ethereum/solidity/pull/11403#discussion_r638950553):

> Please try to avoid output parameters at all costs. Really. At all costs. There is no need for this in C++17.

> This is a rather big function - maybe it's better to have this as a static member in the class?